cpp/oneapi/dal/table/detail/sparse_access_iface.hpp

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/array.hpp"
+
+namespace oneapi::dal::detail {
+namespace v1 {
+
+enum class csr_indexing { zero_based, one_based };
+
+template <typename T>
+struct sparse_block {
+    array<T> data;
+    array<std::int64_t> column_indices;
+    array<std::int64_t> row_indices;
+    csr_indexing indexing;
```

Comment:
Why indexing is needed here?

Reply:
Kirill P. is asked to add it and I agree that it'll be useful in future (when zero_base indexing will be supported)

Reply:
we can support one_based and zero_based format. now we implemented only one_based

Reply:
I do not understand why it is needed to be returned from the accessor. User can ask the table itself which indexing it has. 
Another question: don't we need to ask different indexings from the `pull()` method of the accessor?